jiwer/transforms.py

```diff
@@ -182,12 +184,23 @@ def process_string(self, s: str):
         return s
 
 
-class ReplaceWords(AbstractTransform):
+class SubstituteWords(AbstractTransform):
     def __init__(self, dictionary: Mapping[str, str]):
         self.dictionary = dictionary
 
     def process_string(self, s: str):
         for key, value in self.dictionary.items():
+            s = re.sub(r"\b{}\b".format(re.escape(key)), value, s)
+
+        return s
+
+
+class ApplyRegexSubstitution(AbstractTransform):
```

Comment:
I suggest using `SubstituteRegexes`. This is nice and consistent with `SubstituteWords`.